pkg/cmd/auth/shared/git_credential.go

```diff
@@ -121,7 +142,13 @@ func (flow *GitCredentialFlow) gitCredentialSetup(hostname, username, password s
 }
 
 func gitCredentialHelperKey(hostname string) string {
-	return fmt.Sprintf("credential.%s.helper", strings.TrimSuffix(ghinstance.HostPrefix(hostname), "/"))
+	host := strings.TrimSuffix(ghinstance.HostPrefix(hostname), "/")
+	return fmt.Sprintf("credential.%s.helper", host)
+}
+
+func gistGitCredentialHelperKey(hostname string) string {
```

Comment:
I don't think there's value in this being a separate function. Just pass `gistHost` to `gitCredentialHelperKey`